sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/ParDo.java

```diff
@@ -903,11 +902,53 @@ public String toString() {
     }
   }
 
+  private static String stateDescription(StateSpec<?> spec) {
+    return spec.match(
+        new StateSpec.Cases<String>() {
+          @Override
+          public String dispatchValue(Coder<?> valueCoder) {
+            return "ValueState<" + valueCoder + ">";
```

Comment:
Here, too, maybe using the format of `MoreObjects.toStringHelper(this).add("coder", valueCoder)` will make it all uniform. But since this is really a helper for display data, and `DisplayData` can be structured (somewhat?) it probably makes sense to structure it here using https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/display/DisplayData.java#L145